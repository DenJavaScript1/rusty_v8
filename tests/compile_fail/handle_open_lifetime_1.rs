// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.

pub fn main() {
  let isolate = &mut v8::Isolate::new(mock());
  let context: v8::Local<v8::Context> = mock();
  let scope1 = &mut v8::HandleScope::with_context(isolate, context);

  let global = {
    let local1 = v8::Object::new(scope1);
    v8::Global::new(scope1, local1)
  };

  let local = {
    let scope2 = &mut v8::HandleScope::new(scope1);
    v8::Handle::open(&global, scope2)
  };

  let _ = v8::undefined(scope1);
  assert_eq!(global, local);
}

fn mock<T>() -> T {
  unimplemented!()
}
