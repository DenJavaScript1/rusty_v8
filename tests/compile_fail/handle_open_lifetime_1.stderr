error[E0499]: cannot borrow `*scope1` as mutable more than once at a time
  --> tests/compile_fail/handle_open_lifetime_1.rs:18:25
   |
14 |     let scope2 = &mut v8::HandleScope::new(scope1);
   |                                            ------ first mutable borrow occurs here
...
18 |   let _ = v8::undefined(scope1);
   |                         ^^^^^^ second mutable borrow occurs here
19 |   assert_eq!(global, local);
   |   ------------------------- first borrow later used here
